src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -246,21 +246,23 @@ class MySimulation : public SimulationHelper<simDim>
 
     virtual void pluginUnload()
     {
+        DataConnector &dc = Environment<>::get().DataConnector();
 
         SimulationHelper<simDim>::pluginUnload();
-        __delete(fieldB);
 
-        __delete(fieldE);
+        __delete(myFieldSolver);
+
+        dc.unshare( fieldB->getUniqueId() );
+
+        dc.unshare( fieldE->getUniqueId() );
 
-        __delete(fieldJ);
+        dc.unshare( fieldJ->getUniqueId() );
```

Comment:
Note: currently we need to unshare all buffers else we can get problems with out PMacc singletons (buffer can life longer than we hold the GPU). If we add a clean interface to start/finalize MPI and release the GPU after all buffers, events and streams are destroyed we can skip this unshares.